src/psyclone/psyGen.py

```diff
@@ -6566,4 +6628,73 @@ def __str__(self):
         return "Return[]\n"
 
 
+class Container(Node):
+    '''Node representing something that contains one or more
+    KernelSchedule nodes and/or Container nodes, as well as a name and
+    a SymbolTable. This construct can be used to scope variables,
```

Comment:
I would delete "share variables between KernelSchedules nodes", maybe use something like: "This construct can be used to scope symbols of variables and KernelSchedule names"

Reply:
Done.